webhook/resourcesemantics/validation/validation_admit.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"go.uber.org/zap"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/webhook"
+	"knative.dev/pkg/webhook/resourcesemantics"
+)
+
+var errMissingNewObject = errors.New("the new object may not be nil")
+
+// Callback is a generic function to be called by a consumer of validation
+type Callback struct {
+	Callback func(ctx context.Context, unstructured *unstructured.Unstructured) error
+
+	SupportedVerbs map[admissionv1beta1.Operation]bool
```

Comment:
maybe just make this a list?

Reply:
Additionally - do you mind just aliasing the Operation types? Thus the use of v1beta1 is an implementation detail and we won't need downstream changes when we swap out the API for v1 in the future (ie. next release)

ie. the top level webhook package

```
type Operation = v1beta1.Operation

const (
    Create  Operation = v1beta1.Create
    Update  Operation = v1beta1.Update
    ...
```

Reply:
Aliased into validation.

 I think keeping the map is desirable (really I want a Set) so you can't repeat verbs and makes it easier to not have to search the slice.

Reply:
Sorry moved the alias into the root webhook package. per your suggestion we probably want this everywhere